### ISSUE TITLE
Features req

### ISSUE THREAD

**Issue by Rostam300** [mvrozanti/RAT-via-Telegram#37]
hi good job bro thank to you
Features req:
1: multi victims control it can done by add function to change main token by another token
main toke can have a settoken buttom to redirect rat to another bot for control it
2: remote desktop vnc
thank for your time

**Comment by PulkitSingh256** [comment]
Yes it will be awesome feature if we can acess desktop by vnc for port forwarding we can use ngrok as well

**Comment by Rostam300** [comment]
it no need port forwarding coz pc is vnc server u just need IP and default vnc port 5900 and vnc app its done let's goooo
must important thing is multi victims support

**Comment by PulkitSingh256** [comment]
But the ip must be public and it needs to be configured by router so it would be easier if we forward the port on which vnc server is running so we can acess the vnc through wan

**Comment by Rostam300** [comment]
first focus on multi victims support . must add function to check for new not stored on a file in PC if no new not return settoken on not to add new bot and store and load bot token from file even chat id!

**Comment by mvrozanti** [comment]
We would need to store the token somewhere; It's less hidden than just in the payload.

Why would one want to change the token anyway? It _already is_ able to be multi-controlled (N people can control the same bot). And you can also have N bots as contacts in your own Telegram account. 

What exactly is missing?

**Comment by Rostam300** [comment]
hi . no you cant have more than 20 bot on your telegram acc . and if you can handle all victims from one bot
and add a new bot for each one you dont need to creat new rat with new token! <== it is bad way!
if add that feature you just creat a rat send for all victims and after that it easy to manage . let's think evil
you creat one rat share it ! for each install you receive a new msg on your bot and redirect victim to another
bot for control! even add a /list command to see all victim installed rat to add bot for them .
and one important thing it need a inline keyboard . for easy to use and user friendly 
all them is possible coz when you send a massage to bot all connected devices recive it and return a settoken 😀

**Comment by Rostam300** [comment]
no one work on this project any more?!!!!!

**Comment by badboy051** [comment]
can you add USB spreading&Download files in the host from https://github.com/mehulj94/BrainDamage

**Comment by mvrozanti** [comment]
@badboy051 this is nice. Why do you not use it? On another note; this should/could be added. Problem is USB "spreading" is in no way stealthy or effective since Win>=7 versions don't have any kind of device-insertion-execution like good old `autorun`s. You would not get what you want.

**Comment by mvrozanti** [comment]
@Rostam300  One should be able to "group" bots as I understand, correct? Maybe have one bot responsible for other bots? I'm just worried Telegram finds out about the whole botnet you're thinking about having. Bots connected to bots connected to bots must bring attention.

**Comment by badboy051** [comment]
@mvrozanti I want to use it but it's not fixed and can you please make keyloggers cleaner and remove debug lines, of course, we can remove them but it would be nicer

**Comment by mvrozanti** [comment]
@badboy051 We need proper logging capabilities. Debugging is currently my top priority. Until I get this done I'm not really working on other features. I just need some motivation to do it on the weekend.

Cleaner keylogging file will be a result of logging capabilities.

**Comment by Rostam300** [comment]
mvrozanti hi wellcome back bro , with a abit of changes yes that is Correct
one bot to manage devices than install rat to be easy for control and dont need make new rat for
new one . lets try to make it .
and i have a good idea 
inline keyboard ! make it realy easy to use for ex: click on chande path rat return now send path to me
or ...
where you was i guess you leave this proj

**Comment by mvrozanti** [comment]
I actually did, now I got my setup working again. Is this issue still relevant to your needs?

**Comment by dudeisbrendan03** [comment]
Remote desktop doesn't seem like something that will be easy, as it would normally require port forwarding. The whole point of Telegram is that we don't need to port forward or anything, just deploy and run.

Transferring the rat to a new bot/changing the ID is probably something we can do.

As @mvrozanti said, auto-exec has been removed. The closest vanilla 'auto-exec' we have is when you have a CD in, instead of browsing on a double-click it launches the auto-exec app.

@mvrozanti I can also work on more logging if you want

**Comment by mvrozanti** [comment]
That would be awesome

**Comment by badboy051** [comment]
how about add proxy to bot it self telegram is blocked on some countrys

**Comment by mvrozanti** [comment]
The whole premise of this RAT is that the C&C is not done “directly”, it is done via Telegram

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by badboy051** [comment]
> The whole premise of this RAT is that the C&C is not done “directly”, it is done via Telegram
> 
> Sent with [GitHawk](http://githawk.com)

but in some countries telegram is blocked so its useless but if a socks or https proxy used to connect bot/rat to telegram server it would be great

**Comment by mvrozanti** [comment]
Sadly this program is blocked on countries in which telegram is blocked
